### PR TITLE
unwrap: implement Exception type for better exception handling

### LINKED ISSUE #3130: Provide way to detect fault exceptions returned by `unwrap` package functions
## Is your feature request related to a problem? Please describe.
i'm using https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.102.0/pkg/rpcclient/unwrap and find it conventient. But i found that there is no way to detect fault exception errors:
```go
var ErrNotFound = "smth not found"

// returns [ErrNotFound] if contract throws 'not found' exception.
func getSmth() error {
  _, err := unwrap.Bool(x.invoker.Call(x.contract, method, args...))
  if err != nil {
    if stings.Contains(err.Error(), "not found") {
      return ErrNotFound
    } 
  
    return fmt.Errorf("call %q method of the Neo smart contract %s", method, x.contract)
  }
}

func main() {
  err := getSmth()
  if err != nil {
    if errors.Is(err, ErrNotFound) {
      // specific behavior
    }

    // default behavior
  }
}
```

in this example i base my code logic on the particular string fault exception the contract may throw. The problem is that generic invocation errors (like network or internal) may also contains this string (not this particular example, but in general, e.g. `not found`). So this code is not resistant to potential string collisions and can break logic

currently i use following workaround:
```go
	res, err := x.invoker.Call(x.contract, method, args...)
	if err != nil {
		return fmt.Errorf("call %q method of the Neo smart contract %s", method, x.contract)
	}

	if res.State != vmstate.Halt.String() {
		if strings.Contains(res.FaultException, "not found") {
			return ErrNotFound
		}

		return fmt.Errorf("method %q of the Neo smart contract %s fault exception: %s", method, x.contract, res.FaultException)
	}

	_, err := unwrap.Bool(res, nil)
```

## Describe the solution you'd like
```go
type FaultExceptionError {
  Message string
}

// ...
// Unwap* returns [FaultExceptionError] with fault exception thrown by the
// smart contract when resulting state is not HALT.
func Unwrap*
```
```go
func getSmth() error {
  _, err := unwrap.Bool(x.invoker.Call(x.contract, method, args...))
  if err != nil {
    var faultErr unwrap.FaultExceptionError
    if errors.As(err, &faultErr) && faultErr.Message == "not found" {
      return ErrNotFound
    } 
  
    return fmt.Errorf("call %q method of the Neo smart contract %s", method, x.contract)
  }
}
```

## Describe alternatives you've considered
have no other ideas for now

## Additional context
frequent pattern of unwrapping in the https://github.com/nspcc-dev/neofs-node

### PR DESCRIPTION
**Pull request by roman-khimov**
Fix #3130. "Exception" is used for name since it's shorter and that's the name used in JSON. "VMFault" was also considered as well as "FaultException" (which mirrors result.Invoke).

### CODE REVIEW

File: pkg/rpcclient/unwrap/unwrap.go [hunk 1]
```diff
@@ -25,12 +25,24 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/vm/vmstate"
 )
 
+// Exception is a type used for VM fault messages (aka exceptions). If any of
+// unwrapper functions encounters a FAULT VM state it creates an instance of
+// this type as an error using exception string. It can be used with [errors.As]
+// to get the exact message from VM and compare with known contract-specific
+// errors.
+type Exception string
```
Comment (AnnaShaleva) [inline pkg/rpcclient/unwrap/unwrap.go:33]: > as well as "FaultException" 

`unwrap.FaultException` seems to be more clear than `unwrap.Exception`, don't you think so?

File: pkg/rpcclient/unwrap/unwrap.go [hunk 2]
```diff
@@ -25,12 +25,24 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/vm/vmstate"
 )
 
+// Exception is a type used for VM fault messages (aka exceptions). If any of
+// unwrapper functions encounters a FAULT VM state it creates an instance of
+// this type as an error using exception string. It can be used with [errors.As]
+// to get the exact message from VM and compare with known contract-specific
+// errors.
```
Comment (AnnaShaleva) [inline pkg/rpcclient/unwrap/unwrap.go:32]: So it's not supposed to work with `errors.Is`?

Comment (roman-khimov) [reply pkg/rpcclient/unwrap/unwrap.go:32]: It can't, all instances are different.

File: pkg/rpcclient/unwrap/unwrap.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/rpcclient/unwrap/unwrap.go:33]: Now it's `unwrap.Exception`. I don't mind making it `unwrap.FaultException` if that's better. Most of `unwrap.X` things are to unwrap X from execution results. This one is different in that it's a type. We also have an `Err...` there. I'm open to suggestions of how to make it easier to understand that it's not a regular `unwrap.X` function.

Comment (carpawell) [reply pkg/rpcclient/unwrap/unwrap.go:33]: i would go just `Exception`

**Review by carpawell — APPROVED** [review]

File: pkg/rpcclient/unwrap/unwrap.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/rpcclient/unwrap/unwrap.go:33]: Let's use it.

**Review by AnnaShaleva — APPROVED** [review]

**Review by AliceInHunterland — APPROVED** [review]

**Pull request merged**